polaris-core/src/main/java/org/apache/polaris/core/context/Realm.java

```diff
@@ -29,16 +29,16 @@
  * isolated "universes".
  */
 @PolarisImmutable
-@JsonSerialize(as = ImmutableRealmId.class)
-@JsonDeserialize(as = ImmutableRealmId.class)
-public interface RealmId {
+@JsonSerialize(as = ImmutableRealm.class)
+@JsonDeserialize(as = ImmutableRealm.class)
+public interface Realm {
 
-  static RealmId newRealmId(String id) {
-    return ImmutableRealmId.of(id);
+  static Realm newRealm(String id) {
```

Comment:
suggestion for the sake of clarity: `forId(String id)` or `fromId(String id)`

Reply:
Good point. Resolved it by rename the method.